src/components/subscriptions/hooks/licenseManagerHooks.js

```diff
@@ -0,0 +1,223 @@
+    count: 0,
+    next: null,
+    previous: null,
+  });
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(null);
+
+  const fetch = () => {
+    setIsLoading(true);
+
+    if (!subscriptionUUID) {
+      return;
+    }
+
+    LicenseManagerApiService.fetchSubscriptionUsers(subscriptionUUID, { status, search, page })
+      .then((response) => {
+        setIsLoading(false);
+        setSubscriptionUsers(response.data);
+      })
+      .catch((err) => {
+        NewRelicService.logAPIErrorResponse(err);
+        setIsLoading(false);
+        setError(err);
+      });
+  };
+
+  return {
+    fetch,
+    subscriptionUsers,
+    isLoading,
+    error,
+  };
+};
+
+/*
+This is a helper hook that provides all the subscription data in a single place.
+*/
+export const useSubscriptionData = ({ status, search, page = 1 }) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [errors, setErrors] = useState(null);
+  const [data, setData] = useState({
+    subscriptions: {},
+    overview: {},
+    subscriptionUsers: {},
+  });
+
+  const {
+    fetch: fetchSubscriptions,
+    subscriptions,
+    isLoading: isLoadingSubscriptions,
+    error: subscriptionsError,
+  } = useSubscriptions();
+  const {
+    fetch: fetchSubscriptionUsersOverview,
+    subscriptionUsersOverview,
+    isLoading: isLoadingSubscriptionUsersOverview,
+    error: subscriptionUsersOverviewError,
+  } = useSubscriptionUsersOverview({ subscriptionUUID: subscriptions?.results[0]?.uuid });
+
+  const {
+    fetch: fetchSubscriptionUsers,
+    subscriptionUsers,
+    isLoading: isLoadingSubscriptionUsers,
+    error: subscriptionUsersError,
+  } = useSubscriptionUsers({
+    subscriptionUUID: subscriptions?.results[0]?.uuid,
+    options: { status, search, page },
+  });
+
+  const isInProgress = useMemo(
+    () =>
+      isLoadingSubscriptions || isLoadingSubscriptionUsersOverview || isLoadingSubscriptionUsers,
+    [isLoadingSubscriptions, isLoadingSubscriptionUsersOverview, isLoadingSubscriptionUsers],
+  );
+
+  const hasErrors = useMemo(
+    () => Boolean(subscriptionsError || subscriptionUsersOverviewError || subscriptionUsersError),
+    [subscriptionsError || subscriptionUsersOverviewError || subscriptionUsersError],
+  );
+
+
+  const fetch = () => {
+    // This method will fetch subscriptions and after the successfully fetching subscriptions
+    // data, it will trigger the fetch operation of related data.
+    fetchSubscriptions();
+  };
+
+  useEffect(() => fetchSubscriptionUsersOverview(), [subscriptions]);
+  useEffect(
+    () => fetchSubscriptionUsers(),
+    [subscriptions, status, search, page],
+  );
+
+  useEffect(() => {
+    if (isInProgress) {
+      // Keep the status as isLoading until all requests complete.
+      setIsLoading(true);
+    } else if (hasErrors) {
+      setIsLoading(false);
+      const updatedErrors = {
+        ...(subscriptionsError && { [SUBSCRIPTIONS]: NETWORK_ERROR_MESSAGE }),
```

Comment:
> updatedErrors[SUBSCRIPTIONS] = subscriptionsError && NETWORK_ERROR_MESSAGE; causes an issue as it adds an attribute with null as value. However, I have moved error object construction outside of the function and tried to clean it up a bit.

what about `updatedErrors[SUBSCRIPTIONS] = subscriptionsError ? NETWORK_ERROR_MESSAGE : undefined;`?

Just trying to avoid doing a lot of array spreading here since it makes it a bit less readable.

Reply:
The way it is implemented, code shows all the attributes as error messages even the ones having `undefined` or `null` values. 
One other way would be to check for the value via a guard condition but I think it would not a good solution.

Reply:
OK, don't want to be a blocker on that. Was just looking for potential ways to simplify. We can keep as is!